### PR TITLE
neofetch: remove

### PR DESCRIPTION
**Pull request by AndersonTorres**
By soft abandonment.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by yu-re-ka** [comment]
Maybe recommend an alternative in the alias message?

File: pkgs/top-level/aliases.nix [hunk 1]
```diff
@@ -1186,6 +1186,7 @@ mapAliases ({
   ncat = throw "'ncat' has been renamed to/replaced by 'nmap'"; # Converted to throw 2022-02-22
   neap = throw "neap was removed from nixpkgs, as it relies on python2"; # Added 2022-01-12
   neochat = libsForQt5.kdeGear.neochat; # added 2022-05-10
+  neofetch = throw "neofetch was removed: upstream had not updated it since 2021-12-10. Seek for alternative fetchers, like afetch."; # Added 2023-09-16
```
Comment (wegank) [inline pkgs/top-level/aliases.nix:1189]: Looks like the version of `afetch` in nixpkgs is even older than `neofetch`?

Comment (AndersonTorres) [reply pkgs/top-level/aliases.nix:1189]: Seriously?
I was thinking in a minimal, C-only or shell-only fetcher, but the one I found is older?

Is there any options at all?

Comment (AndersonTorres) [reply pkgs/top-level/aliases.nix]: Let's go with hyfetch

**Comment by Gerg-L** [comment]
How about we don't just straight up remove one of the most well-known and used linux packages
at the very least have a warning for a while before throwing

**Comment by Gerg-L** [comment]
following the train of logic of ignoring code usage and only worrying about code age and maintenance we should remove almost every single `Xorg` package

**Comment by NotAShelf** [comment]
this PR is solely redundant. we don't need to decide what people should be using so aggressively

neofetch works and a lot of people will instinctively put it in their configs, a throw saying "um actually you shouldn't" will just be a thorn in their boot

**Comment by eclairevoyant** [comment]
https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+in%3Atitle+neofetch I don't see this to be a massive maintenance concern either, since I only see one issue listed that was addressed

**Comment by AndersonTorres** [comment]
> following the train of logic of ignoring code usage and only worrying about code age and maintenance we should remove almost every single `Xorg` package

Except Xorg was updated less than a month ago:
https://www.x.org/releases/individual/

**Comment by eclairevoyant** [comment]
> Xorg was updated less than a month ago: https://www.x.org/releases/individual/

Many Xorg utilities haven't seen updates in over a decade. Some as long as 15 years ago.

**Comment by AndersonTorres** [comment]
> this PR is solely redundant. we don't need to decide what people should be using so aggressively

I thought on the side of Nixpkgs maintenance. We are using patches over the tip of master, after all.

> 
> neofetch works and a lot of people will instinctively put it in their configs, a throw saying "um actually you shouldn't" will just be a thorn in their boot

That's a good argument. I will close this, then.

**Comment by Gerg-L** [comment]
> Many Xorg utilities haven't seen updates in over a decade. Some as long as 15 years ago.

this is what i meant
> That's a good argument. I will close this, then.
thanks

**Pull request closed without merging**